### PR TITLE
fix(datastore): Fix bundle size regression caused by lodash import changes

### PR DESCRIPTION
**Pull request by scottn12**
## Description

This PR fixes the bundle size regression that was caused by the changes in the lodash import syntax. 

The syntax was initially changed to accommodate the new node 16 module resolution (https://github.com/microsoft/FluidFramework/pull/18826). The syntax it was changed to caused the entire lodash library to be loaded. This PR fixes this by replacing cloneDeep with structuredClone

### CODE REVIEW

File: packages/runtime/datastore/src/localChannelContext.ts [hunk 1]
```diff
@@ -3,9 +3,8 @@
  * Licensed under the MIT License.
  */
 
-import lodashPkg from "lodash";
-// eslint-disable-next-line @typescript-eslint/unbound-method
-const { cloneDeep } = lodashPkg;
+// eslint-disable-next-line import/no-unresolved
+import cloneDeep from "lodash.clonedeep";
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/runtime/datastore/src/localChannelContext.ts:7]: i wonder if we can just remove lodash from this package. not sure if there are things besides clone used in this package.

[the remark this answers is not in the document]
Comment (scottn12) [reply packages/runtime/datastore/src/localChannelContext.ts:7]: Do you think we could replace cloneDeep with structuredClone 1:1? If so we can remove the lodash dependency entirely

[the remark this answers is not in the document]
Comment (scottn12) [reply packages/runtime/datastore/src/localChannelContext.ts:7]: > i wonder if we can just remove lodash from this package. not sure if there are things besides clone used in this package.

Sorry, for some reason only your first comment appeared for me when I sent my last response. I'll try this out

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -17045,7 +17025,7 @@ packages:
   /@fluidframework/protocol-base/2.0.2:
     resolution: {integrity: sha512-9AbQ8FuZ5UNZgOWmNaWxs7CH08H9qRYX6OkTUbRXNjdJaFRSCBCkPSA23aHD85EN49UEkkpfoyH1SsQchHuR4Q==}
     dependencies:
-      '@fluidframework/common-utils': 3.0.0
+      '@fluidframework/common-utils': 3.1.0
```
Comment (markfields) [inline pnpm-lock.yaml:17028]: Not sure what's in this bump, but would be better to avoid it I think.  Could be ok though since it's minor.

Comment (markfields) [reply pnpm-lock.yaml:17028]: Like maybe just manually go revert these lines

File: packages/runtime/datastore/package.json [hunk 3]
```diff
@@ -128,7 +128,6 @@
 		"@fluidframework/runtime-definitions": "workspace:~",
 		"@fluidframework/runtime-utils": "workspace:~",
 		"@fluidframework/telemetry-utils": "workspace:~",
-		"lodash": "^4.17.21",
```
Comment (tylerbutler) [inline packages/runtime/datastore/package.json:131]: Don't you need to add the @ungap/structured-clone dependency?

Comment (scottn12) [reply packages/runtime/datastore/package.json:131]: CI seems to be passing without it. Any thoughts @zhenmichael or @anthony-murphy?

Comment (noencke) [reply packages/runtime/datastore/package.json:131]: I guess structuredClone is a built-in for browsers for the last couple of years. I wasn't aware.

Comment (scottn12) [reply packages/runtime/datastore/package.json:131]: Sounds like loop has had issues with structuredClone, so I included the @ungap dependency in the latest commit

**Review by ChumpChief — APPROVED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (scottn12) [reply pnpm-lock.yaml:17028]: Wouldn't the 3.1 version still be installed for anyone consuming this release though regardless of the lock file?

**Pull request merged**